src/odemis/driver/keysight.py

```diff
@@ -620,9 +704,31 @@ def _parseMessage(self, msg):
         elif re.match(r"sour[1-2]:freq?", msg):
             channel = int(msg[4])
             self._sendAnswer(b"%+.15E" % (self._output_states[channel].frequency,))
-        elif re.match(r"sour[1-2]:appl:(sin|squ|ramp|puls)", msg):
+        elif re.match(r"sour[1-2]:appl:(sin|squ|ramp|puls|dc)", msg):
```

Comment:
you can utilize the regex with group names to do the variable assignment for you.
Something in the lines of:
```
elif match := re.match(r"sour(?P<channel>[1-2]):appl:(?P<waveform>sin|squ|ramp|puls|dc)(?: (?P<params>.+))?", msg):
    channel = int(match.group("channel"))
    waveform = match.group("waveform")
    params = match.group("params").split(",") if match.group("params") else []
```

Reply:
I'll not implement this for now. I just want to keep this change simple. Adding "dc" is enough.